packages/localize/src/LocalizeManager.js

```diff
@@ -495,32 +495,30 @@ export class LocalizeManager {
     }
     if (this._autoLoadOnLocaleChange) {
       this._loadAllMissing(newLocale, oldLocale);
+      this.loadingComplete.then(() => {
+        this.dispatchEvent(new CustomEvent('localeChanged', { detail: { newLocale, oldLocale } }));
+      });
+    } else {
+      this.dispatchEvent(new CustomEvent('localeChanged', { detail: { newLocale, oldLocale } }));
     }
-    this.dispatchEvent(new CustomEvent('localeChanged', { detail: { newLocale, oldLocale } }));
   }
 
   /**
    * @param {string} newLocale
    * @param {string} oldLocale
-   * @returns {Promise.<Object>}
    * @protected
    */
   _loadAllMissing(newLocale, oldLocale) {
     const oldLocaleNamespaces = this.__storage[oldLocale] || {};
     const newLocaleNamespaces = this.__storage[newLocale] || {};
-    /** @type {Promise<Object|void>[]} */
-    const promises = [];
     Object.keys(oldLocaleNamespaces).forEach(namespace => {
       const newNamespaceData = newLocaleNamespaces[namespace];
       if (!newNamespaceData) {
-        promises.push(
-          this.loadNamespace(namespace, {
-            locale: newLocale,
-          }),
-        );
+        this.loadNamespace(namespace, {
```

Comment:
calling loadNamespace will ensure the promise is added to the __namespaceLoaderPromisesCache which enables the user to then await the loaders through `localize.loadingComplete`. Therefore, it is not necessary here to store those promises separately and return them, loadingComplete should be used for this instead as it uses the central promises cache